src/aiperf/server_metrics/manager.py

```diff
@@ -308,6 +309,15 @@ async def _stop_all_collectors(self) -> None:
             except Exception as e:
                 self.error(f"Failed to stop collector for {endpoint_url}: {e}")
 
+    async def _delayed_shutdown(self) -> None:
+        """Shutdown service after a delay to allow command response to be sent.
+
+        Waits before calling stop() to ensure the command response
+        has time to be published and transmitted to the SystemController.
+        """
+        await asyncio.sleep(Environment.SERVER_METRICS.SHUTDOWN_DELAY)
```

Comment:
I know this is probably a quick fix, but it seems like a magic number kind of situation.

Is there any way the shutdown can wait on the actual condition that needs to exist? Like detecting that the command response has been published and transmitted?

Reply:
@matthewkotila its tricky with zmq. One of the true fixes for this is to fix the system controller to detect and allow optional services to die (and stop waiting for a response)

Reply:
that is a larger thing, which I had worked on at one point and never got to finishing it. Partially because it was linked with the macOS fixes and changes with multiprocessing, that I reverted because the approach was too slow